File: android/app/src/main/java/com/reactnativebridgegradient/nativeComonents/RNGradientViewManager.java
package com.reactnativebridgegradient;

import android.support.annotation.Nullable;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;
public class RNGradientViewManager extends SimpleViewManager<RNGradientView> {
    @Override
    public String getName() {
        return "RNGradientView";
    }
    @Override
    protected RNGradientView createViewInstance(ThemedReactContext reactContext) {
        return new RNGradientView(reactContext);
    }
    // Properties
    @ReactProp(name = "progress")
    public void setProgress(RNGradientView view, @Nullable float progress) {
        view.setProgress(progress);
    }
    @ReactProp(name = "cornerRadius")
    public void setCornerRadius(RNGradientView view, @Nullable float cornerRadius) {
        view.setCornerRadius(cornerRadius);
    }
    @ReactProp(name = "fromColor", customType = "Color")
    public void setFromColor(RNGradientView view, @Nullable int color) {
        view.setFromColor(color);
    }
    @ReactProp(name = "toColor", customType = "Color")
    public void setToColor(RNGradientView view, @Nullable int color) {
        view.setToColor(color);
    }
}

File: ios/NativeComponents/RNGradientViewManager.m
//
//  RNGradientViewManager.m
//  ReactNativeBridgeGradient
//
//  Created by proSolvo on 2/13/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "RNGradientViewManager.h"
#import "RNGradientView.h"

@implementation RNGradientViewManager

RCT_EXPORT_MODULE()

- (UIView *)view {
  return [[RNGradientView alloc] init];
}

RCT_EXPORT_VIEW_PROPERTY(progress, NSNumber);
RCT_EXPORT_VIEW_PROPERTY(cornerRadius, NSNumber);
RCT_EXPORT_VIEW_PROPERTY(fromColor, UIColor);
RCT_EXPORT_VIEW_PROPERTY(toColor, UIColor);


@end
